pkl-gradle/src/test/kotlin/org/pkl/gradle/TestsTest.kt

```diff
@@ -112,34 +112,33 @@ class TestsTest : AbstractTest() {
     assertThat(output.trimStart())
       .startsWith(
         """
-      > Task :evalTestGatherImports
-
-      > Task :evalTest FAILED
-      pkl: TRACE: 8 = 8 (file:///file, line x)
-      module test
-        facts
-          ✅ sum numbers
-          ✅ divide numbers
-          ❌ fail
-             4 == 9 (file:///file, line x)
-             "foo" == "bar" (file:///file, line x)
-        examples
-          ✅ user 0
-          ✅ user 1 #0
-          ❌ user 1 #1
-             (file:///file, line x)
-             Expected: (file:///file, line x)
-             new {
-               name = "Parrot"
-               age = 35
-             }
-             Actual: (file:///file, line x)
-             new {
-               name = "Welma"
-               age = 35
-             }
-      ❌ 66.7% tests pass [2/6 failed], 66.7% asserts pass [3/9 failed]
-    """
+        > Task :evalTestGatherImports
+
+        > Task :evalTest FAILED
+        pkl: TRACE: 8 = 8 (file:///file, line x)
+        module test
+          facts
+            ✅ sum numbers
+            ✅ divide numbers
+            ❌ fail
+               4 == 9 (file:///file, line x)
+               "foo" == "bar" (file:///file, line x)
+          examples
+            ✅ user 0
+            ❌ user 1
+               (file:///file, line x)
+               Expected: (file:///file, line x)
+               new {
+                 name = "Parrot"
+                 age = 35
+               }
+               Actual: (file:///file, line x)
+               new {
+                 name = "Welma"
+                 age = 35
+               }
```

Comment:
IMHO this is a "regression", in  #498, I added the "breaking out examples" to make testing examples (e.g. large example files) easier. An example can have "different facts", each object / part of the example becomes a fact and when it fails, you can quickly see that it's the Nth part of the example file.

This PR goes back to making "examples opaque", a failure in an examples file doesn't let you know how many other parts pass and doesn't give you info of which part of the example breaks 😐

Reply:
Gotcha, I think it makes sense that you might want some details about _which_ example failed.

How about an output that looks like this? This would still be considered an assertion inside the test, but this makes it clearer _which_ assertion failed.

```
examples
  ✅ user 0
  ❌ user 1
    #1:
      (file:///file, line x)
      Expected: (file:///file, line x)
      new {
        name = "Parrot"
        age = 35
      }
      Actual: (file:///file, line x)
      new {
        name = "Welma"
        age = 35
      }
    #2:
      (file:///file, line x)
      Expected: (file:///file, line x)
      5
      Actual: (file:///file, line x)
      10
```

Reply:
Awesome, IMHO that's actually much better than my output, as it's closer to the `facts` output: it's green if all the assertions/examples pass and red when at least one fails, and then it shows the errors of the assertions / "example cases" that failed 👌 

Also, re. whitespace and alignment, one minor nit is I'd make sure everything is tabulated to the `user`. The output you pasted was "one-space-less" than what I'd use 😄.

Here's what I'd prefer :
```
examples
  ✅ user 0
  ❌ user 1
     #1:
       (file:///file, line x)
       Expected: (file:///file, line x)
       new {
         name = "Parrot"
         age = 35
       }
       Actual: (file:///file, line x)
       new {
         name = "Welma"
         age = 35
       }
     #2:
       (file:///file, line x)
       Expected: (file:///file, line x)
       5
       Actual: (file:///file, line x)
       10
```

Finally, and this is something that I didn't do in my PR but maybe you can consider doing, is to remove one of the `file:///file, line x` from the example's error output, because I think it's duplicated in `Actual`. That way you save one line per error as well 😅 :
```
examples
  ✅ user 0
  ❌ user 1
     #1:
       Expected: (file:///file, line x)
       new {
         name = "Parrot"
         age = 35
       }
       Actual: (file:///file, line x)
       new {
         name = "Welma"
         age = 35
       }
     #2:
       Expected: (file:///file, line x)
       5
       Actual: (file:///file, line x)
       10
```
 
 And If you decide not to remove it, you could also put it next to the `#<ID>`:
 ```
 examples
  ✅ user 0
  ❌ user 1
     #1: (file:///file, line x)
       Expected: (file:///file, line x)
       new {
         name = "Parrot"
         age = 35
       }
       Actual: (file:///file, line x)
       new {
         name = "Welma"
         age = 35
       }
     #2: (file:///file, line x)
       Expected: (file:///file, line x)
       5
       Actual: (file:///file, line x)
       10
```

Reply:
It's not duplicated, actually! There's three source sections to show:

1. Within the test
2. Within `pkl-expected.pcf`
3. Within `pkl-actual.pcf`

But good idea re: putting it next to the counter.

Reply:
Aaaah true, true, you are probably right, I was just thinking it would duplicate it in the case where examples are defined in-file, in which case I hought `Actual` would match with the `#<nth>` example.